dev/com.ibm.ws.threading_policy_fat/test-applications/basicfat/src/web/PolicyExecutorServlet.java

```diff
@@ -5656,4 +5656,85 @@ public void testExecute() throws Exception {
         executor.shutdownNow();
     }
 
+    //Test the getRunningTaskCount method
+    @Test
+    public void testGetRunningTaskCount() throws Exception {
+        PolicyExecutor executor = provider.create("testgetRunningTaskCount").maxConcurrency(2);
+
+        CountDownLatch beginLatch1 = new CountDownLatch(1);
+        CountDownLatch continueLatch1 = new CountDownLatch(1);
+        CountDownLatch beginLatch2 = new CountDownLatch(1);
+        CountDownLatch continueLatch2 = new CountDownLatch(1);
+
+        assertEquals(0, executor.getRunningTaskCount()); //no tasks running
+
+        CountDownTask task1 = new CountDownTask(beginLatch1, continueLatch1, TimeUnit.HOURS.toNanos(1));
+        Future<Boolean> future1 = executor.submit(task1); //task1 will begin running and block on continueLatch1
+        assertTrue(beginLatch1.await(TIMEOUT_NS, TimeUnit.NANOSECONDS));
+
+        assertEquals(1, executor.getRunningTaskCount()); //task1 is running
+
+        CountDownTask task2 = new CountDownTask(beginLatch2, continueLatch2, TimeUnit.HOURS.toNanos(1));
+        Future<Boolean> future2 = executor.submit(task2); //task2 will begin running and block on continueLatch2
+        assertTrue(beginLatch2.await(TIMEOUT_NS, TimeUnit.NANOSECONDS));
+
+        assertEquals(2, executor.getRunningTaskCount()); //task1 and task2 are running
+
+        continueLatch1.countDown(); //allow task1 to complete
+        assertTrue(future1.get(TIMEOUT_NS, TimeUnit.NANOSECONDS));
+        assertEquals(1, executor.getRunningTaskCount()); //task2 is running
```

Comment:
Not reliable to check the running count after Future.get returns because there is a small window where the state is updated that would allow Future.get to return just prior to decrement of the run count.  You would need to poll for count to reach desired value within a reasonable amount of time.

Reply:
Good catch, I'll update the test